apiserver/modelmanager/modelmanager.go

```diff
@@ -839,9 +847,35 @@ func (c *ModelManagerAPI) UnsetModelDefaults(args params.UnsetModelDefaults) (pa
 	}
 
 	for i, arg := range args.Keys {
-		results.Results[i].Error = common.ServerError(
-			c.state.UpdateModelConfigDefaultValues(nil, arg.Keys, arg.CloudTag, arg.CloudRegion),
-		)
+		if arg.CloudRegion != "" {
+			rspec, err := c.makeRegionSpec(arg.CloudTag, arg.CloudRegion)
+			if err != nil {
+				results.Results[i].Error = common.ServerError(
+					errors.Trace(err))
+				continue
+			}
+			results.Results[i].Error = common.ServerError(
+				c.state.UpdateModelConfigDefaultValues(nil, arg.Keys, rspec),
+			)
+		} else {
+			results.Results[i].Error = common.ServerError(
```

Comment:
I think it'd be a little neater to de-dupe these two calls, and do the conditional rspec thing like you did in the other method.